mllib/src/main/scala/org/apache/spark/ml/util/Identifiable.scala

```diff
@@ -19,15 +19,24 @@ package org.apache.spark.ml.util
 
 import java.util.UUID
 
+
 /**
- * Object with a unique id.
+ * Trait for an object with an immutable unique ID that identifies itself and its derivatives.
  */
-private[ml] trait Identifiable extends Serializable {
+trait Identifiable {
+
+  /**
+   * An immutable unique ID for the object and its derivatives.
```

Comment:
Good question! I don't think the `Identifiable` interface is user facing, not developer facing either. The only useful function to developers is the `randomUID` under `object Identifiable`.